src/overlord/Proxy.js

```diff
@@ -307,7 +307,14 @@ export class ProxyOverlord {
         console.log("Proxy finished any pending caching transactions");
     }
 
-    _remoteCall(commandOrString) {
+    // count - the number of collapsed requests
+    allCollapsed(path, count) {
+        const allRequests = count + 1; // including the parent request
+        let options = {'Overlord-request-path': path, 'Overlord-active-requests-count': allRequests};
+        return this._remoteCall("/waitActiveRequests", options);
+    }
```

Comment:
I made this method 'sync' so that the promise could be passed to blockSendingUntil(). Other _remoteCall() callers could be probably simply waited for with 'await'.

Reply:
This is not my area of expertise, but I was under impression that the _caller_ can use `async foo() { await promise(); }` just like it can use `foo() { return promise(); }`.  In other words, there is no reason _not_ to make this method `async` even if some callers do not want to `await` its result.

Reply:
You a right - an async returns a native Promise. However, after adjusting this I faced another problem:

```
Test attempt failure (1/1): TypeError: externalEvent.tap is not a function
```

meaning that the Promise returned by that async method lacks tap(). I added a Promise.resolve() wrap to fix this, as described [here](https://stackoverflow.com/questions/44158629/get-bluebird-promise-from-async-await-functions/44166729#44166729).